### PR TITLE
FUSETOOLS2-2458 - Provide contextual menu to transform routes

### PR DESCRIPTION
**Pull request by hdamarcelo**
- Add tranform commands to the "New Camel file" contextual menu
- Right clicking a file provides de ability to transform it to XML/Yaml DSLs
- Right clicking a folder tries to transform all routes in it to XML/Yaml DSLs

### CODE REVIEW

**Comment by hdamarcelo** [comment]
Peek 2024-10-03 02-30

**Review by apupier — APPROVED** [review]
looks good.
Please add tests either in this PR or in another PR as another iteration

File: src/ui-test/tests/commands.transform.test.ts [hunk 1]
```diff
@@ -16,11 +16,11 @@
...
+import { ActivityBar, DefaultTreeSection, EditorView, InputBox, Menu, SideBarView, VSBrowser, ViewItem, WebDriver, Workbench } from 'vscode-uitests-tooling';
 import * as pjson from '../../../package.json';
-import { EXAMPLE_TRANSFORM_COMMAND_JAVA_FILE, EXAMPLE_TRANSFORM_COMMAND_XML_FILE, EXAMPLE_TRANSFORM_COMMAND_YAML_FILE, FOLDER_WITH_RESOURCES_FOR_TRANSFORM_COMMAND, TRANSFORM_ROUTES_IN_FOLDER_TO_YAML_COMMAND_ID, TRANSFORM_ROUTE_TO_YAML_COMMAND_ID, deleteFile, killTerminal, openFileInEditor, waitUntilEditorIsOpened, waitUntilExtensionIsActivated } from '../utils/testUtils';
+import { EXAMPLE_TRANSFORM_COMMAND_JAVA_FILE, EXAMPLE_TRANSFORM_COMMAND_XML_FILE, EXAMPLE_TRANSFORM_COMMAND_YAML_FILE, FOLDER_WITH_RESOURCES_FOR_TRANSFORM_COMMAND, NEW_CAMEL_FILE_LABEL, TRANSFORM_CAMEL_ROUTE_YAML_DSL_LABEL, TRANSFORM_ROUTES_IN_FOLDER_TO_YAML_COMMAND_ID, TRANSFORM_ROUTE_TO_YAML_COMMAND_ID, deleteFile, killTerminal, openFileInEditor, waitUntilEditorIsOpened, waitUntilExtensionIsActivated } from '../utils/testUtils';
 
-describe('Transform Camel Routes to YAML using commands', function () {
+describe.only('Transform Camel Routes to YAML using commands', function () {
```
Comment (apupier) [inline src/ui-test/tests/commands.transform.test.ts:23]: forgot the only

File: src/ui-test/utils/testUtils.ts [hunk 2]
```diff
@@ -594,3 +601,30 @@ export async function initNewCamelFile(type: string, name: string, kamelet: stri
...
+
+/**
+ * Opens the context menu for a given route in the sidebar.
+ * @param route The route for which the context menu should be opened.
+ * @returns A promise that resolves to the opened ContextMenu.
+ */
+export async function openContextMenu(route: string): Promise<ContextMenu> {
+    const item = await (await new SideBarView().getContent().getSection('explorer')).findItem(route) as ViewItem;
+    const menu = await item.openContextMenu();
+    return menu;
```
Comment (apupier) [inline src/ui-test/utils/testUtils.ts:613]: can directly return the menu

**Pull request merged**